spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -255,8 +292,10 @@ public MessageGroup removeMessageFromGroup(Object groupId, Message<?> messageToR
 				SimpleMessageGroup group = this.groupIdToMessageGroup.get(groupId);
 				Assert.notNull(group, "MessageGroup for groupId '" + groupId + "' " +
 						"can not be located while attempting to remove Message from the MessageGroup");
-				group.remove(messageToRemove);
-				group.setLastModified(System.currentTimeMillis());
+				if (group.remove(messageToRemove)) {
+					this.groupToUpperBound.get(groupId).release();
```

Comment:
Perhaps add a utility method for all these releases, and add an `Assert.state()` that the UB is not null.